### PR TITLE
Add poll_write docs

### PR DESCRIPTION
**Pull request by quininer**
Close https://github.com/tokio-rs/tls/issues/58

### CODE REVIEW

**Comment by djc** [comment]
(Asked a bunch of questions in #58.)

File: tokio-rustls/src/lib.rs [hunk 1]
```diff
@@ -1,4 +1,35 @@
...
+//!
+//! Most TLS implementations will have an internal buffer to improve throughput,
+//! and rustls is no exception.
+//!
+//! When we write data to `TlsStream`, we always write rustls buffer first,
+//! then take out rustls encrypted data packet, and write it to data channel (like TcpStream).
+//! When data channel is pending, some data may remain in rustls buffer.
+//! at this point we must call flush to write again it into data channel.
+//!
+//! `tokio-rustls` To keep it simple and correct, [TlsStream] will behave like `BufWriter`.
```
Comment (djc) [inline tokio-rustls/src/lib.rs:13]: I find this hard to understand. I feel like there are three steps here that should be plainly stated:

1. When or in what kind of situations this problem arises ("Does this apply to me")
2. Context on the underlying problem; there is some of this, but as written it's hard to relate to as a library user
3. What to do to solve the issue (that is, in what kind of situations should I be calling `poll_flush()` explicitly)

File: tokio-rustls/src/lib.rs [hunk 2]
```diff
@@ -1,4 +1,35 @@
...
+//!
+//! ## Why don't we write during `poll_read`?
+//!
+//! We did this in the early days of `tokio-rustls`, but it caused some bugs.
+//! We can solve these bugs through some solutions, but this will cause performance degradation (reverse false wakeup).
+//!
+//! And reverse write will also prevent us implement full duplex in the future.
+//!
+//! see <https://github.com/tokio-rs/tls/issues/40>
```
Comment (djc) [inline tokio-rustls/src/lib.rs:22]: I don't understand who the intended audience for this section is; that is, which (kind of) library users care about this problem?

File: tokio-rustls/src/lib.rs [hunk 3]
```diff
@@ -1,4 +1,35 @@
...
+//!
+//! ## Why can't we handle it like `native-tls`?
+//!
+//! native-tls also needs to call flush, but it will be much less.
+//!
+//! When data channel returns to pending, `native-tls` will falsely report the number of bytes it consumes,
+//! which does not conform to convention of `AsyncWrite` trait.
+//! This means that if you give inconsistent data in two `poll_write`, it may cause unexpected behavior.
```
Comment (djc) [inline tokio-rustls/src/lib.rs:30]: Here, too, it's not super clear to me who would ask this kind of question or why it's relevant.

File: tokio-rustls/src/lib.rs (hunk 1 above)
Comment (quininer) [reply tokio-rustls/src/lib.rs:13]: Sorry, I think it is common sense that `BufWriter` needs to call `flush`. so I subconsciously did not give more explanation.

File: tokio-rustls/src/lib.rs (hunk 2 above)
Comment (quininer) [reply tokio-rustls/src/lib.rs:22]: People often ask me why `tokio-rustls` needs to call `flush` but `tokio-native-tls` does not. why does `tokio-rustls` not write data during `poll_read`. these are real user questions.

**Pull request merged**